refactor prepXfer and postXfer

Pyngon opened the pull request:
Refactor prepXfer and postXfer by moving the createIOR, createIOV and data copy to prepXfer.

Also, added some error handling during failure to clean up handle and io objects.

Pyngon committed:
refactor prepXfer and postXfer

zackyoray commented on src/plugins/hf3fs/hf3fs_backend.cpp:
```diff
@@ -155,13 +134,27 @@ nixl_status_t nixlHf3fsEngine::postXfer (const nixl_xfer_op_t &operation,
+    if (handle == nullptr) {
+        hf3fs_handle = new nixlHf3fsBackendReqH();
+        handle_created = true;
+    } else {
```
checked this and this is not valid, lets always create new handle

Pyngon replied:
OK

zackyoray commented on src/plugins/hf3fs/hf3fs_backend.cpp:
```diff
@@ -194,27 +192,63 @@ nixl_status_t nixlHf3fsEngine::postXfer (const nixl_xfer_op_t &operation,
+                                       const nixl_opt_b_args_t* opt_args)
+{
+    // Handle null pointer case - should have been initialized in prepXfer
+    if (handle == nullptr) {
```
As we discussed we can remove this

zackyoray commented on src/plugins/hf3fs/hf3fs_backend.cpp:
```diff
@@ -194,27 +192,63 @@ nixl_status_t nixlHf3fsEngine::postXfer (const nixl_xfer_op_t &operation,
+        hf3fs_handle->io_list.push_back(io);
+    }
+
+    hf3fs_handle->status = NIXL_HF3FS_STATUS_PREPARED;
```
regarding this status, so i think we shuold or use in start of the funciton (for example quick exit in prepXfer) or remove this at all.

Pyngon replied:
From the comment above, we will always create a new handle, i think we can remove this.

zackyoray commented on src/plugins/hf3fs/hf3fs_backend.cpp:
```diff
@@ -194,27 +192,63 @@ nixl_status_t nixlHf3fsEngine::postXfer (const nixl_xfer_op_t &operation,
+    // Clean up previously created IOs in the list
+    for (auto prev_io : hf3fs_handle->io_list) {
+        delete prev_io;
+    }
```
new line is needed after `}` for passing pre-commit

Pyngon committed:
checkXfer to handle properly when postXfer could be called multiple times

Also, cleanup io object during when handle get reused in prepXfer
fd will be dereg in deregisterMem, so do not have to dereg in releaseReqH

Pyngon committed:
prepXfer handle is always valid, remove check

Also, remove status from nixlHf3fsBackendReqH as not needed
